### PR TITLE
Cleanup trafficgen_entry.sh, Dockerfile, README 

### PR DESCRIPTION
**Pull request by jianzzha**
Also, fix a "send_teaching_warmup" handling bug; add "no_promisc" boolean to the API.

### CODE REVIEW

File: rfc2544/README.md [hunk 1]
```diff
@@ -15,26 +15,53 @@ The container image build from this directory can be used for either automation
...
+First, identify the numa node associated with the NIC pic slot,
 ```
-CMD ["/root/trafficgen_entry.sh", "start"]
+cat /sys/bus/pci/devices/<pci_address>/numa_node
 ```
 
-Run the trafficgen
-`podman run -it --rm --privileged -v /dev:/dev -v /sys:/sys -v /lib/modules:/lib/modules --cpuset-cpus 4-11 -e pci_list=0000:03:00.0,0000:03:00.1 docker.io/cscojianzhan/trafficgen`
+Select a cpuset on this numa node.
+
```
Comment (akiselev1) [inline rfc2544/README.md:69]: I would add cpuset selection logic and helpful commands here. Assuming our device with <pci_address> is on NUMA node1:
```
# lscpu | grep node1
NUMA node1 CPU(s):   1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79
# cat /sys/devices/system/cpu/isolated
4-19
```
That points to isolated CPUs 5,7,9,11,13,15,17,19 as available for the trafficgen.

**Review by dkosteck — APPROVED** [review]
Changes look good, I tested them on my setup and it worked well.

File: rfc2544/README.md [hunk 2]
```diff
@@ -15,26 +15,53 @@ The container image build from this directory can be used for either automation
...
+For manual test, users normally do not use pod, but run the trafficgen container directly,
+```
+podman run -it --rm --privileged -v /dev/hugepages:/dev/hugepages -v /sys/bus/pci/devices:/sys/bus/pci/devices -v /lib/modules:/lib/modules --cpuset-cpus 4-11 -e pci_list=0000:03:00.0,0000:03:00.1 localhost/trafficgen start
```
Comment (akiselev1) [inline rfc2544/README.md]: We should probably use` --cpuset-cpus 5,7,9,11,13,15` in this command to illustrate our selection.

File: rfc2544/README.md [hunk 3]
```diff
@@ -44,29 +71,25 @@ How to install the DDP packet can be found in Intel's E810 DDP package release n
 
 If using VF on the E810 as the trafficgen ports, then the extra mount for the DDP package is not required.
 
-The TRex version also make a different on E810. With the 2.88 TRex version, the E810 PF works but VF does not; with 3.02 TRex version, the E810 VF works but PF does not.
- 
+The TRex version makes a different on E810. With the 2.88 TRex version, the E810 PF works but VF does not; with 3.02 TRex version, the E810 VF works but PF does not.
```
Comment (akiselev1) [inline rfc2544/README.md]: "makes a difference"

Comment (dkosteck) [reply rfc2544/README.md]: Good catch

File: rfc2544/README.md (hunk 1 above)
Comment (jianzzha) [reply rfc2544/README.md:69]: @akiselev1 See if the explanation looks ok

Comment (akiselev1) [reply rfc2544/README.md:69]: Looks good.

**Review by akiselev1 — APPROVED** [review]

**Comment by jianzzha** [comment]
@akiselev1 minor change, updated from 5 cores for 7 cores for better performance

**Comment by akiselev1** [comment]
Confirmed, line rate is possible with 7 cores.

**Pull request merged**